test/HalResourceTest.php

```diff
@@ -235,13 +239,22 @@ public function testWithElementProxiesToEmbedIfResourceCollectionValueProvided()
         $this->assertEquals(['foo' => $collection], $new->getElements());
     }
 
-    public function testWithElementDoesNotProxyToEmbedIfAnEmptyArrayValueIsProvided(): void
+    public function testWithElementWillEmbedAnEmptyArrayIfAnEmptyArrayValueIsProvided(): void
     {
-        $resource = new HalResource(['foo' => 'bar']);
+        $resource = new HalResource(['foo' => 'bar'], [], [], true);
         $new      = $resource->withElement('bar', []);
 
         $representation = $new->toArray();
-        $this->assertEquals(['foo' => 'bar', 'bar' => []], $representation);
+        self::assertSame(['foo' => 'bar', '_embedded' => ['bar' => []]], $representation);
+    }
+
+    public function testWithElementDoesNotProxyToEmbedIfNullValueIsProvided(): void
```

Comment:
```suggestion
    public function testWithElementDoesNotProxyToEmbedIfNullValueIsProvidedAndEmbedEmptyCollectionsEnabled(): void
```

Same remark as previous.